rt/frontend/jaxrs/src/main/java/org/apache/cxf/jaxrs/impl/UriBuilderImpl.java

```diff
@@ -661,6 +662,14 @@ private void setUriParts(URI uri) {
         }
     }
 
+    private boolean queryIsEmpty() {
+        return query == null || query.isEmpty();
+    }
+
+    private boolean sspMatchessPath(final URI uri) {
+        return uri.getRawSchemeSpecificPart() != null && uri.getRawSchemeSpecificPart().equals("//" + uri.getPath());
```

Comment:
It would be good to check if `uri.getPath()` is not null, otherwise `"//" + uri.getPath()` becomes something like `//null`